sklearn/ensemble/tests/test_iforest.py

```diff
@@ -347,3 +347,12 @@ def test_n_features_deprecation():
 
     with pytest.warns(FutureWarning, match="`n_features_` was deprecated"):
         est.n_features_
+
+
+def test_iforest_with_n_jobs():
+    """This is a non-regression test for PR #23252"""
+    X, _ = make_classification(n_samples=50000, n_features=100)
```

Comment:
For me, the bug segfault was triggered around here:

```suggestion
    X, _ = make_classification(n_samples=85_000, n_features=100, random_state=0)
```

I think the bug triggers when joblib's [Parallel](https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html) uses memmapped, which is triggered when the data is big enough.